src/components/Carousel/Carousel.stories.tsx

```diff
@@ -94,19 +98,153 @@ const Scroll_Story: ComponentStory<typeof Carousel> = (args) => (
   <Carousel {...args} />
 );
 
+type scrollVisibilityApiType = React.ContextType<typeof VisibilityContext>;
+
+// When using custom buttons, manually handle touch via scrollVisibilityApiType and apiRef.
+const onWheel = (
+  apiObj: scrollVisibilityApiType,
+  event: React.WheelEvent
+): void => {
+  const isTouchpad =
+    Math.abs(event?.deltaX) !== 0 || Math.abs(event?.deltaY) < 15;
+
+  if (isTouchpad) {
+    event?.stopPropagation();
+    return;
+  }
+
+  if (event?.deltaY < 0) {
+    apiObj?.scrollNext();
+  } else if (event.deltaY > 0) {
+    apiObj?.scrollPrev();
+  }
+};
+
+const Scroll_Custom_Buttons_Story: ComponentStory<typeof Carousel> = (args) => {
+  const htmlDir: string = useCanvasDirection();
+  const apiRef: React.MutableRefObject<autoScrollApiType> = useRef(
+    {} as scrollVisibilityApiType
+  );
+
+  const [nextDisabled, setNextDisabled] = useState<boolean>(false);
+  const [previousDisabled, setPreviousDisabled] = useState<boolean>(true);
+
+  const scrollPrevAsync = async (): Promise<void> => {
+    await apiRef.current?.scrollPrev();
+    setNextDisabled(apiRef.current.isLastItemVisible);
+    setPreviousDisabled(apiRef.current.isFirstItemVisible);
+  };
+
+  const scrollNextAsync = async (): Promise<void> => {
+    await apiRef.current?.scrollNext();
+    setNextDisabled(apiRef.current.isLastItemVisible);
+    setPreviousDisabled(apiRef.current.isFirstItemVisible);
+  };
+
+  return (
+    <Card bordered height={440} style={{ padding: 48 }} width={400}>
+      <div
+        style={{
+          height: '100%',
+          left: htmlDir === 'rtl' ? 'unset' : 8,
+          position: 'absolute',
+          right: htmlDir === 'rtl' ? 8 : 'unset',
+          top: 0,
+          width: 28,
+        }}
+      >
+        <Button
```

Comment:
are this previous button supposed to be outside of the carousel? Kind of weird structure
```ts
<PreviousButton />
<Carousel>
   <NextButton />
</Carousel>
```

Reply:
Yes, because it's your own button. That's the point.

Reply:
It's not weird. One of Eightfold apps' [main carousel deps - react-horizontal-scrolling-menu](https://www.npmjs.com/package/react-horizontal-scrolling-menu) provides the same exact option.